capstone/src/main/webapp/script.js

```diff
@@ -336,31 +328,63 @@ async function averagePrecinctSentiment(policePrecinct) {
   const sentimentAverage = sentimentCount/list.length;
   return getSentimentColor(sentimentAverage);
 }
-
+//colors precinct on map based on strong dislike to strong like 
 function getSentimentColor(averageFeelings) {
+  let precinctColor = '#228B22';
   if (Math.round(averageFeelings) == 5) {
-    return '#165B33';
+    precinctColor = '#165B33';
   } else if (Math.round(averageFeelings) == 4) {
-    return '#146B3A';
+    precinctColor = '#146B3A';
   } else if (Math.round(averageFeelings) == 3) {
-    return '#F8B229';
+    precinctColor = '#F8B229';
   } else if (Math.round(averageFeelings) == 2) {
-    return '#EA4630';
+    precinctColor = '#EA4630';
   } else {
-    return '#BB2528';
+    precinctColor = '#BB2528';
   }
+  return precinctColor;
 }
-
+//global variable for precinctLayer data layer and checkbox selection 
 const mapAndSelection = {};
+//following two functions are only used in index.html 
 /* eslint-disable no-unused-vars */
+//unchecks boxes when new checkbox is checked 
 function onlyOne(checkbox) {
   const checkboxes = document.getElementsByName('check');
   checkboxes.forEach((item) => {
-    if (item !== checkbox) item.checked = false;
+    if (item !== checkbox) {
+      item.checked = false;
+    }
   });
-  mapAndSelection.selection = checkbox.id;
-  drawCheckboxLayer();
+
+  if (mapAndSelection.selection == checkbox.id) {
+    mapAndSelection.selection = 'noneSelected';
+    resetMap();
+  } else {
+    mapAndSelection.selection = checkbox.id;
+    drawCheckboxLayer();
+  }
+}
+//goes to log in page if user is not logged in 
+async function showStats() {
+  const logStatus = await fetch('/status');
+  const status = await logStatus.json();
+  if (status == true) {
+    window.location.replace('statistics.html');
+  } else {
+    window.location.replace('/login');
```

Comment:
I think the better approach here is to check the status of the user in the backend once they navigate to statistics page.

This means that the Java Servlet will be responsible for checking the user's permissions or otherwise will redirect to the login page. Doing this in the frontend means, I can still manually edit my url to navigate to the statistics page.